qiime2/sdk/plugin_manager.py

```diff
@@ -38,29 +38,57 @@ def iter_entry_points(cls):
 
     # This class is a singleton as it is slow to create, represents the
     # state of a qiime2 installation, and is needed *everywhere*
-    def __new__(cls):
+    def __new__(cls, add_plugins=True):
```

Comment:
we should add an `else` branch to complain about `add_plugins=False` if the singleton already exists.

Reply:
I might not have understood your request, check out https://github.com/qiime2/qiime2/pull/511/files#diff-81b344e767b72470d737b8c2972a9ffeR46-R50

Reply:
That looks right (the final diff at least, the link is being weird for me)!